lib/filewatch/tail.rb

```diff
@@ -11,13 +11,11 @@ class Tail
     attr_writer :target
 
     def self.new_observing(opts = {})
-      new.tap do |instance|
-        instance.target = ObservingTail.new(opts)
-      end
+      new({}, ObservingTail.new(opts))
     end
 
-    def initialize(opts = {})
-      @target = YieldingTail.new(opts)
+    def initialize(opts = {}, target = YieldingTail.new(opts))
```

Comment:
Feels weird to have a scalar value after an options/keyword-style argument here. My feeling is that `def initialize(target=..., opts={})` is ... better somehow, though I cannot quite articulate why I feel this way.